### PR TITLE
Docstrings 1.0

### PR DESCRIPTION
**Pull request by adamgayoso**
I'm leaving differential expression out for now as we can do the docs when refactoring the interface.

### CODE REVIEW

File: scvi/core/models/__init__.py [hunk 1]
```diff
@@ -32,16 +32,42 @@
...
+        test_size: Optional[float] = None,
+        lr: float = 1e-3,
+        n_epochs_kl_warmup: int = 400,
+        n_iter_kl_warmup: int = None,
+        frequency: Optional[int] = None,
+        train_fun_kwargs: dict = {},
         **kwargs
     ):
+        """
+        Train the model.
```
Comment (romain-lopez) [inline scvi/core/models/__init__.py:46]: This needs to be more descriptive. "Skeleton class for training model"? Contains the primitive methods for training a variational autoencoder?

File: scvi/core/models/__init__.py [hunk 2]
```diff
@@ -67,26 +93,85 @@ def train(
...
+    def get_elbo(
+        self,
+        adata: Optional[AnnData] = None,
+        indices: Optional[Sequence[int]] = None,
+        batch_size: int = 128,
+    ) -> float:
+        """
+        Return the ELBO for the data.
 
+        The ELBO is the lower bound on the log likelihood of the data used for optimization
```
Comment (romain-lopez) [inline scvi/core/models/__init__.py:105]: the elbo is a lower bound

File: scvi/core/models/__init__.py [hunk 3]
```diff
@@ -143,17 +230,16 @@ class RNASeqMixin:
     @torch.no_grad()
```
Comment (romain-lopez) [inline scvi/core/models/__init__.py:230]: The class must have a description as well. Contains all the primitives for xxxx

File: scvi/core/models/__init__.py [hunk 4]
```diff
@@ -620,7 +724,7 @@ def get_latent_library_size(
 
 
 class BaseModelClass(ABC):
```
Comment (romain-lopez) [inline scvi/core/models/__init__.py:726]: Same here, this should be commented so we know what this is doing

File: scvi/core/models/__init__.py [hunk 5]
```diff
@@ -733,7 +843,20 @@ def _get_init_params(self, locals):
         }
         return user_params
 
-    def save(self, dir_path, overwrite=False):
+    def save(self, dir_path: str, overwrite: bool = False):
+        """
+        Save the state of the `scvi` model.
+
+        The trainer optimizer state nor the trainer history are saved.
```
Comment (romain-lopez) [inline scvi/core/models/__init__.py:850]: neither ..nor

File: scvi/core/models/__init__.py [hunk 6]
```diff
@@ -752,7 +875,30 @@ def save(self, dir_path, overwrite=False):
             pickle.dump(user_attributes, f)
 
     @classmethod
-    def load(cls, adata: AnnData, dir_path, use_cuda=False):
+    def load(cls, adata: AnnData, dir_path: str, use_cuda: bool = False):
+        """
+        Create an `scvi` model from the saved output.
```
Comment (romain-lopez) [inline scvi/core/models/__init__.py:880]: Instantiate xxx

File: scvi/core/models/__init__.py [hunk 7]
```diff
@@ -32,16 +32,42 @@
 class VAEMixin:
```
Comment (romain-lopez) [inline scvi/core/models/__init__.py:32]: Global question. Is it normal this is in a __init__ file? I would expect that's not good practice. Should we have a specific file for these primitives instead?

File: scvi/dataset/_anndata.py [hunk 8]
```diff
@@ -32,7 +32,7 @@ def get_from_registry(adata: anndata.AnnData, key: str) -> np.array:
 
     Returns
     -------
-    np.array containing the data requested
+    The requested data
```
Comment (romain-lopez) [inline scvi/dataset/_anndata.py:35]: what type is it?

File: scvi/models/gimvi.py [hunk 9]
```diff
@@ -239,7 +280,38 @@ def get_imputed_values(
...
-    def load(cls, adata_seq: AnnData, adata_spatial: AnnData, dir_path, use_cuda=False):
+    def load(
+        cls,
+        adata_seq: AnnData,
+        adata_spatial: AnnData,
+        dir_path: str,
+        use_cuda: bool = False,
+    ):
+        """
+        Create an `scvi` model from the saved output.
```
Comment (romain-lopez) [inline scvi/models/gimvi.py:291]: gimvi?

File: scvi/models/gimvi.py [hunk 10]
```diff
@@ -239,7 +280,38 @@ def get_imputed_values(
...
+            AnnData organized in the same way as data used to train model.
+            AnnData must be registered via :func:`~scvi.dataset.setup_anndata`.
+        dir_path
+            Path to saved outputs.
+        use_cuda
+            Whether to load model on GPU.
+
+        Returns
+        -------
+        `scvi` model
```
Comment (romain-lopez) [inline scvi/models/gimvi.py:308]: gimvi

**Review by romain-lopez — COMMENTED** [review]
I flagged a couple of things, but looks good otherwise

File: scvi/core/models/__init__.py (hunk 3 above)
Comment (adamgayoso) [reply scvi/core/models/__init__.py:230]: can add these in the docstrngs for the backend PR

File: scvi/core/models/__init__.py (hunk 4 above)
Comment (adamgayoso) [reply scvi/core/models/__init__.py:726]: same

File: scvi/core/models/__init__.py (hunk 7 above)
Comment (adamgayoso) [reply scvi/core/models/__init__.py:32]: I've seen it before, but we should evaluate the organization of everything after all the huge changes are in place.

File: scvi/dataset/_anndata.py (hunk 8 above)
Comment (adamgayoso) [reply scvi/dataset/_anndata.py:35]: the type of the function signature gets placed in the doc

**Pull request merged**